File: backend/utils/currency.py
# utils/currency.py

# Tỷ giá cố định để chuyển đổi từ BRL sang VND
BRL_TO_VND_RATE = 5200  # 1 BRL = 5,200 VND

def brl_to_vnd(amount_brl):
    """
    Chuyển đổi số tiền từ BRL sang VND
    
    Args:
        amount_brl (float): Số tiền BRL cần chuyển đổi
        
    Returns:
        float: Số tiền tương ứng bằng VND
    """
    if amount_brl is None:
        return None
    
    return round(amount_brl * BRL_TO_VND_RATE)

def format_vnd(amount_vnd):
    """
    Format số tiền VND theo định dạng tiền tệ Việt Nam
    
    Args:
        amount_vnd (float): Số tiền VND cần format
        
    Returns:
        str: Chuỗi tiền tệ đã được format
    """
    if amount_vnd is None:
        return "0 ₫"
    
    return f"{int(amount_vnd):,} ₫".replace(",", ".")

File: frontend/src/utils/currency.js
/**
 * Định dạng số tiền theo định dạng tiền tệ Việt Nam (VND)
 * @param {number} amount - Số tiền cần định dạng
 * @returns {string} Chuỗi đã định dạng theo tiền VND
 */
export const formatVND = (amount) => {
    if (amount === null || amount === undefined) {
      return "0 ₫";
    }
    
    // Định dạng số theo chuẩn Việt Nam: dấu chấm phân cách hàng nghìn
    return new Intl.NumberFormat('vi-VN', {
      style: 'currency',
      currency: 'VND',
      minimumFractionDigits: 0,
      maximumFractionDigits: 0
    }).format(amount);
  };
  
  /**
   * Chuyển đổi từ BRL sang VND (chỉ sử dụng nếu API chưa chuyển đổi)
   * @param {number} amountBRL - Số tiền BRL cần chuyển đổi
   * @returns {number} Số tiền tương ứng bằng VND
   */
  export const convertBRLtoVND = (amountBRL) => {
    if (amountBRL === null || amountBRL === undefined) {
      return 0;
    }
    
    const EXCHANGE_RATE = 5200; // 1 BRL = 5,200 VND
    return Math.round(amountBRL * EXCHANGE_RATE);
  };